dipy/core/gradients.py

```diff
@@ -244,3 +244,55 @@ def gradient_table(bvals, bvecs=None, big_delta=None, small_delta=None,
                                            small_delta=small_delta,
                                            b0_threshold=b0_threshold,
                                            atol=atol)
+
+
+def reorient_bvecs(gtab, rots):
+    """
+    Reorient the directions in a GradientTable according
+
+    When correcting for motion, rotation of the diffusion-weighted volumes
+    might cause systematic bias in rotationally invariant measures, such as FA
+    and MD, and also cause characteristic biases in tractography, unless the
+    gradient directions are appropriatel reoriented to compensate for this
+    effect [Leemans2009]_.
+
+    Parameters
+    ----------
+    gtab : a GradientTable class instance.
+        The nominal gradient table with which the data were acquired.
+    rots : (n, 3) array.
+        The x, y, z rotations (in radians) associated with the motion
+        parameters for each of the volumes in the experiment.
+
+    Returns
+    -------
+    gtab : a GradientTable class instance with the reoriented directions
+
+
+    References
+    ----------
+    .. [Leemans2009] The B-Matrix Must Be Rotated When Correcting for
+       Subject Motion in DTI Data. Leemans, A. and Jones, D.K. (2009).
+       MRM, 61: 1336-1349
+    """
+    new_bvecs = gtab.bvecs[~gtab.b0s_mask]
```

Comment:
Are you sure you wan to just apply this to only `~gtab.b0s_mask`?

Either way, there should error checking here to make sure the number rotations matches the number of gradients. When the data has b0s mixed in with gradients (for example hc data) it's easy to occidentally pass in too many or too few rotation angles.